### DISCUSSION TITLE
vale realmente a pena começar pelo Front End em 2021?

### DISCUSSION THREAD

**Discussion by AlyssonPsilva** [frontendbr/forum#2112] [Discussão]
Bom dia, boa tarde ou boa noite pessoal, sou novo aqui no github e já estudo front end, HTML e CSS e JS, a alguns meses e, agora que já estou em um nível... sei lá me considero em um nível bom já que consigo aplicar qualquer design utilizando somente essas linguagens sem sofrer muito, comecei a procurar trabalhos, freelancers e a pesquisar vagas de emprego em pequenas empresas, porém o que eu vi foi que as empresas não querem saber se você aprendeu realmente html css e js e sim se você sabe usar alguma biblioteca como Bootstrap ou o Tailwind e isso me desanimou pra caramba porquê eu me dediquei muito pra aprender cada parte de HTML e CSS e de repente todo esse conhecimento é irrelevante e eu particularmente não gosto de usar essas ferramentas pois elas deixam o código semanticamente mal feito, feio e mal organizado enchendo o html de classes e divs que fica parecendo até style inline, e por eu não gostar de usar isso sei lá estou pensando em partir pro Back-end onde parece que o conhecimento bruto tem maior valor e simplesmente esquecer tudo que aprendi sobre o front. então, resolvi abrir esse tópico de discussão para ver opiniões do pessoal da comunidade e fomentar uma discussão saudável sobre esse tema.  então pessoal, pra um iniciante será que vale realmente a pena se dedicar ao front end mesmo que o conhecimento seja resumido a simplesmente essas ferramentas que citei para as empresas?

**Comment by luanbernardes** [comment]
Fala Alysson, vejo muitas empresas pequenas assim mesmo mas vc pode encarar com uma porta de entrada para conseguir algo mais legal e com certeza tem muita empresa legal precisando de gente boa, mas o começo pode ser meio chato mesmo. Na minha opiniao toda (quase todas hahaha) libs/frameworks tem o seu espaço, Bootstrap por exemplo ajuda a fazer uma estrutura responsiva e/ou uma layout basico muito mais rápido, uma lib sólida com anos no mercado também podemos dar uma olhada como resolveram problemas. 

Minha sugestão é voce dar uma olhada nas vagas/empresas que vc gosta, veja oq elas pedem tecnicamente e faz um plano de estudo para chegar lá, não precisa ser a risca que aquela vaga quer, por exemplo a vaga pede React mas vc pode escolher 1 framework para estudar. 

Agora questão de mercado de trabalho front-end, vejo que está cada vez melhor (tenho aproximadamente 8 anos na área Frontend) 

Em modo geral voce acha muita vaga tanto para front quanto pra back, tenho amigos escolhendo vaga com um bom salario e em projetos bons e empresas dahora nos dois caminhos (back e front). Vejo bastante vagas full stack tbm caso goste dos dois lados da forca hehehe
Um detalhe importante, não é em 6 meses que se aprende a escrever um bom código, seja HTML, CSS, JS...
Não sei se ja conheçe mas para ter uma ideia de caminhos: https://roadmap.sh/frontend

**Comment by anotte** [comment]
Ola Jose, tudo bom?

Entre em contato comigo, temos oportunidades para vc por aqui sim!
***@***.***

[]'s
Alexandre Notte
EWTI
[55 11] 98335.3833

**Comment by ninetails** [comment]
Opa Alysson

Encare frameworks e bibliotecas como ferramentas de meio e não final. Elas auxiliam no desenvolvimento ágil pois evitam que você reinvente a roda. Infelizmente sim, tem bibliotecas que "sujam" o código e deixam ele menos semântico, mas toda a bagagem que você adquiriu até agora servirá de base de, mesmo limitado pelos frameworks, tentar fazer da melhor forma possível.

Realmente, eu também não gosto de frameworks css, também acho que eles não geram um código semântico... Mas você pode tentar estudar e posteriormente vagas com outros frameworks mais gerais de frontend relacionados a componentização, como os mainstreams React, Angular e Vue. Como estes você ainda controla o que vai para o front, talvez fique menos limitado e menos enojado que usando frameworks de desenvolvimento rápido (como geralmente são os de CSS com componentes já prontos e muitas vezes não muito bem estruturados).

Mas o que você adquiriu de conhecimento não foi em vão. Ainda há muitos na comunidade que ainda advocam por semântica, acessibilidade e boas práticas. Só começar a ir nos eventos que sempre vai ter uma palestra dessas :D

Aliás... quais são suas pretensões? Você quer trabalhar com Frontend? Quer trabalhar com TI? Quer trabalhar com backend? Já cogitou mobile? Ou quer ir pela grana? Ou apenas estar empregado?

Dependendo destas respostas, a carreira é tua. A escolha também é tua. Digital está aí e vagas é o que não falta (não confundir com que seria fácil entrar em qualquer uma delas). Qualquer campo de Digital está em alta em 2021, que mesmo com final de pandemia, o comportamento do consumidor médio ainda existe e cresce em estar atrás de uma telinha interativa.

**Reply by AlyssonPsilva** [reply]
> Opa Alysson
> 
> Encare frameworks e bibliotecas como ferramentas de meio e não final. Elas auxiliam no desenvolvimento ágil pois evitam que você reinvente a roda. Infelizmente sim, tem bibliotecas que "sujam" o código e deixam ele menos semântico, mas toda a bagagem que você adquiriu até agora servirá de base de, mesmo limitado pelos frameworks, tentar fazer da melhor forma possível.
> 
> Realmente, eu também não gosto de frameworks css, também acho que eles não geram um código semântico... Mas você pode tentar estudar e posteriormente vagas com outros frameworks mais gerais de frontend relacionados a componentização, como os mainstreams React, Angular e Vue. Como estes você ainda controla o que vai para o front, talvez fique menos limitado e menos enojado que usando frameworks de desenvolvimento rápido (como geralmente são os de CSS com componentes já prontos e muitas vezes não muito bem estruturados).
> 
> Mas o que você adquiriu de conhecimento não foi em vão. Ainda há muitos na comunidade que ainda advocam por semântica, acessibilidade e boas práticas. Só começar a ir nos eventos que sempre vai ter uma palestra dessas :D
> 
> Aliás... quais são suas pretensões? Você quer trabalhar com Frontend? Quer trabalhar com TI? Quer trabalhar com backend? Já cogitou mobile? Ou quer ir pela grana? Ou apenas estar empregado?
> 
> Dependendo destas respostas, a carreira é tua. A escolha também é tua. Digital está aí e vagas é o que não falta (não confundir com que seria fácil entrar em qualquer uma delas). Qualquer campo de Digital está em alta em 2021, que mesmo com final de pandemia, o comportamento do consumidor médio ainda existe e cresce em estar atrás de uma telinha interativa.

sim meu único problema com esses "frameworks CSS" é que eles sujam o código, sei lá eu não consigo ver aquilo no código e me sentir bem em falar que fui eu que fiz, acho que eu sou meio perfeccionista demais, se não fosse isso eu não teria problema em usar eles. ah e respondendo a sua pergunta, eu não sei bem se vou ficar no desenvolvimento web, ou mesmo no front end. eu quero experimentar todas as áreas e ver qual eu gosto mais. eu quero aprender um pouco de cada área, não tudo obviamente, mas no futuro eu quero ter o conhecimento de por qualquer projeto em prática, seja um site, um sistema de gestão ou até um IOT de geladeira kkkk pode parecer meio louco e vai dar muito trabalho, mas eu quero saber pelo menos o intermediário de cada área. eu só estou começando pelo front end web pois parece ser o meio mais simples de ganhar um dinheiro pra se estabilizar na carreira.

**Reply by ninetails** [reply]
Entendi, se vc tem essa oportunidade pra estudar de tudo, estude :D

Mas mesmo assim, ainda acho que tem carreira pra tudo de TI rs

Ainda acho que quem usa "frameworks CSS" é pq quer um desenvolvimento rápido ao custo da semântica, e faz sentido para agências e produtoras, só tentar algo que não seja isso :)

Aconselharia ver então frameworks de componentização, que vc tem o controle do código que será mandado pro navegador (digo, código visível gerado de output, como HTML e CSS e tal)

**Comment by fiuzagr** [comment]
@AlyssonPsilva vale a pena começar a desenvolver, seja qual for a frente pela qual vc quer iniciar. Você vai encontrar boas oportunidades em frontend web ou mobile, ou em backend, ou tbm como devops. É mais fácil e recomendável focar em uma frente para entrar no mercado, assim como você fez. (Frontend não é mais fácil que backend, isso é um grande mito)

No front web, HTML, CSS e JS é o início e muito importante. Você precisa de um conhecimento sólido disso. Mas em nenhuma empresa vc vai trabalhar com isso puramente. Todas elas vão solicitar frameworks/libs que abstraem muitas coisas e permitem acelerar o desenvolvimento.

Sua visão do backend é errada. Vc não vai encontrar empresas desenvolvendo em backend sem framework/lib. E você vai ter os mesmos problemas que levantou com os frameworks de front: vai ter estruturas engessadas, vai ter semântica adicional, vai ter "mágica" (aquelas coisas que está implícita na abstração e não ficam claras), vai ter que aprender muitas coisas adicionais para usar somente com aquele framework/lib.

Outra coisa, o que você aprendeu de front nunca mais será perdido. Você não deveria esquecer isso caso opte por mudar de frente. Pelo contrário, vc deveria usar isso para se destacar dos demais. O termo fullstack está distorcido sim, mas é ainda o que as empresas mais precisam, uma vez que os times estão cada vez mais autônomos e precisando de pessoas que se dão bem em qualquer frente.

Como você já escolheu frontend, eu recomendo vc aprender uma stack muito bem. React e Vue são as mais fáceis de achar em startups e empresas muito boas. O Angular é tbm uma boa opção, mas está bem mais fraco comparado ao React. (Eu digo stack pq vc não vai aprender somente uma coisa e sim todo um ecossistema para acompanhar a lib escolhida)

Sobre os frameworks CSS, esquece eles. Se a vaga pedir, diz que sabe e vai pra cima, pq com uma boa base de HTML e CSS, vc aprende rápido qualquer um deles. (E eu posso lhe garantir que a maioria das empresas não sabem nem usa-los)

É importante lembrar que a área de frontend está evoluindo muito rápido. Não faz sentido não aprender frameworks. Se você quer ficar na área, escolha uma stack e se aprofunde nela. Depois de um tempo, busque outras para diversificar e ampliar seu conhecimento.

Perca o preciosismo de querer códigos sempre perfeitos. Num ambiente real você não vai conseguir mante-los sempre no melhor estado, por vários fatores.

**Comment by carlosblanc0** [comment]
Cara, eu acho que vale. Se pergunte o seguinte: você se vê fazendo algo chato por muito tempo? Então melhor focar em algo que curte e gere renda.
